src/Tests/Microsoft.NET.TestFramework/AssemblyInfo.cs

```diff
@@ -12,6 +12,45 @@ namespace Microsoft.NET.TestFramework
 {
     public static class AssemblyInfo
     {
+        public static List<string> GetParameterlessAttributes(string assemblyPath)
+        {
+            var parameterlessAttributes = new List<string>();
+
+            using (var stream = File.OpenRead(assemblyPath))
+            using (var peReader = new PEReader(stream))
+            {
+                var metadataReader = peReader.GetMetadataReader();
+                var assemblyDefinition = metadataReader.GetAssemblyDefinition();
+
+                foreach (var handle in assemblyDefinition.GetCustomAttributes())
+                {
+                    var attribute = metadataReader.GetCustomAttribute(handle);
+                    var constructor = metadataReader.GetMemberReference((MemberReferenceHandle)attribute.Constructor);
+                    var type = metadataReader.GetTypeReference((TypeReferenceHandle)constructor.Parent);
+                    var name = metadataReader.GetString(type.Name);
+
+                    var signature = metadataReader.GetBlobReader(constructor.Signature);
+                    var value = metadataReader.GetBlobReader(attribute.Value);
+                    var header = signature.ReadSignatureHeader();
+
+                    const ushort prolog = 1; // two-byte "prolog" defined by ECMA-335 (II.23.3) to be at the beginning of attribute value blobs
+                    if (value.ReadUInt16() != prolog || header.Kind != SignatureKind.Method || header.IsGeneric)
+                    {
+                        throw new BadImageFormatException();
+                    }
+
+                    var paramCount = signature.ReadCompressedInteger();
+                    if (paramCount <= 0 || // must have at least 1 parameter
+                        signature.ReadSignatureTypeCode() != SignatureTypeCode.Void) // return type must be void
```

Comment:
The comment in the first line here is no longer correct.  And is the check for void return type correct here?  The previous code was skipping attributes that didn't have parameters OR that weren't void.  Now we want parameterless attributes, but it seems like maybe we still want to skip them if they aren't void returns.  (I don't know the metadata format at all, so I'm not sure.)

Reply:
Removed the comment and the void check now